app/src/main/java/co/nimblehq/alpine/sample/CameraCaptureActivity.kt

```diff
@@ -85,11 +115,86 @@ class CameraCaptureActivity : ComponentActivity() {
         )
         // Attach the viewFinder's surface provider to Preview use case
         preview?.setSurfaceProvider(binding.pvCameraCaptureViewFinder.surfaceProvider)
+    }
+
+    private fun updateCameraUi() {
+        binding.ibCameraCapture.setOnClickListener {
+            loadingDialog.show()
+            imageCapture?.let {
+                val photoFile = createFile(outputDirectory)
+                val outputOptions = ImageCapture.OutputFileOptions.Builder(photoFile)
+                    .build()
+
+                // Setup image capture listener which is triggered after photo has been taken
+                it.takePicture(
+                    outputOptions,
+                    cameraExecutor,
+                    object : ImageCapture.OnImageSavedCallback {
+                        override fun onError(exc: ImageCaptureException) {
+                            showErrorMessage()
+                        }
+
+                        override fun onImageSaved(output: ImageCapture.OutputFileResults) {
+                            val savedUri = output.savedUri ?: Uri.fromFile(photoFile)
+                            notifyOnNewPhotoAdded(savedUri)
+                            photoCaptured.value = photoFile
+                        }
+                    })
+            }
+        }
+    }
 
+    private fun observeCapture() {
+        lifecycleScope.launch {
+            photoCaptured
+                .catch {
+                    showErrorMessage()
+                }
```

Comment:
Not entirely sure what error we're supposed to be catching here? 🤔

Reply:
You got the point, I am not sure as well, but I saw in the dee-money project, they catched the error so I just added it here. 
Let me remove it